### ISSUE TITLE
error occurs sometimes - no object returned

### ISSUE THREAD

**Issue by hartenthaler** [JustCarmen/webtrees-fancy-imagebar#37]
Thank you for making this nice module available again!

`Return value of JustCarmen\Webtrees\Module\FancyImagebarModule::getLinkedObject() must be an object, none returned …/modules_v4/jc-fancy-imagebar/FancyImagebarModule.php:535
#0 …/modules_v4/jc-fancy-imagebar/FancyImagebarModule.php(326): JustCarmen\Webtrees\Module\FancyImagebarModule->getLinkedObject(Object(Fisharebest\Webtrees\Media))
#1 …/modules_v4/jc-fancy-imagebar/FancyImagebarModule.php(209): JustCarmen\Webtrees\Module\FancyImagebarModule->fancyImagebar()
#2 …/resources/views/layouts/default.phtml(190): JustCarmen\Webtrees\Module\FancyImagebarModule->bodyContent()
#3 [internal function]: Fisharebest\Webtrees\View::{closure}(Object(JustCarmen\Webtrees\Module\FancyImagebarModule), '_jc-fancy-image...')
#4 …/vendor/illuminate/support/Collection.php(1120): array_map(Object(Closure), Array, Array)
#5 …/resources/views/layouts/default.phtml(191): Illuminate\Support\Collection->map(Object(Closure))
#6 …/app/View.php(189): include('/mnt/web312/a3/...')
#7 …/app/View.php(286): Fisharebest\Webtrees\View->render()
#8 …/app/Helpers/functions.php(203): Fisharebest\Webtrees\View::make('layouts/default', Array)
#9 …/app/Http/ViewResponseTrait.php(58): view('layouts/default', Array)
#10 …/app/Http/RequestHandlers/MediaPage.php(96): Fisharebest\Webtrees\Http\RequestHandlers\MediaPage->viewResponse('media-page', Array)
#11 …/app/Http/Middleware/WrapHandler.php(83): Fisharebest\Webtrees\Http\RequestHandlers\MediaPage->handle(Object(Nyholm\Psr7\ServerRequest))
#12 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\WrapHandler->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
`

**Comment by JustCarmen** [comment]
Thanks for reporting. I think I need to have an escape for orphan images (images not linked to any object). This has to do with the new feature where the images are clickable and link to the individual, family or source page.

**Comment by fisharebest** [comment]
FYI - the function `getLinkedObject` has a typehint of `object`.

`object` isn't available until PHP 7.2, and webtrees 2.0 supports PHP 7.1

(In webtrees 2.1, we will raise the minimum PHP to 7.2).

**Comment by fisharebest** [comment]
I had a quick look at the code.

If no linked record is found, `function getLinkedObject` will reach the end without returning anything.

https://github.com/JustCarmen/webtrees-fancy-imagebar/blob/2.0.7/FancyImagebarModule.php#L518

Suggestion: Typehint the return value as `?GedcomRecord` (i.e. `GedcomRecord` or `null`).

Tip: the `linkedXxx` functions return a `Collection` object, which is like an array with extra features.

So, you can use `$media->linkedIndividuals('OBJE')->first()` which returns the first entry, or `null`.

For performance, you could do something like this, which will stop after the first query that has a result.

```
return
    $media->linkedIndividuals('OBJE')->first() ?? 
    $media->linkedfamilies('OBJE')->first() ??
    $media->linkedSourcess('OBJE')->first();
```

Finally, at this line, you need to check for `null`, rather an empty string

https://github.com/JustCarmen/webtrees-fancy-imagebar/blob/2.0.7/FancyImagebarModule.php#L412
